### PR TITLE
drivers : spi : mec1501 : XEC SPI driver

### PR DESCRIPTION
**Pull request by scottwcpg**
SPI driver for MEC1501 QMSPI supporting synchronous only.

Signed-off-by: Scott Worley <scott.worley@microchip.com>

### CODE REVIEW

**Comment by scottwcpg** [comment]
Pushed update to fix Kconfig issue.

File: drivers/spi/Kconfig.xec_qmspi [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Kconfig - Atmel SAM0 SERCOM SPI
```
Comment (dcpleung) [inline drivers/spi/Kconfig.xec_qmspi]: Please change the description (or remove it if you want).

File: drivers/spi/spi_xec_qmspi.c [hunk 2]
```diff
@@ -0,0 +1,682 @@
+/*
+ * Copyright (c) 2019 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(spi_xec);
```
Comment (dcpleung) [inline drivers/spi/spi_xec_qmspi.c]: Please use LOG_MODULE_REGISTER(spi_xec, CONFIG_SPI_LOG_LEVEL) instead, and remove the LOG_LEVEL #define above.

File: soc/arm/microchip_mec/mec1501/Kconfig.soc [hunk 3]
```diff
@@ -70,4 +70,4 @@ config SOC_MEC1501_EXT_32K_PARALLEL_CRYSTAL
 config SOC_MEC1501_VTR3_1_8V
 	bool "VTR3 power rail is tied to 1.8V"
 	help
-	  Set this is if VTR3 power sourcejumper in the board is changed.
+	  Set this is if VTR3 power sourcejumper in the board is changed.
```
Comment (dcpleung) [inline soc/arm/microchip_mec/mec1501/Kconfig.soc]: This seems to have no change except the line return at the end.

**Review by dcpleung — COMMENTED** [review]
LGTM, except a few minor things.

File: drivers/spi/Kconfig.xec_qmspi (hunk 1 above)
Comment (scottwcpg) [reply drivers/spi/Kconfig.xec_qmspi]: Done.

File: drivers/spi/spi_xec_qmspi.c (hunk 2 above)
Comment (scottwcpg) [reply drivers/spi/spi_xec_qmspi.c]: Done

File: soc/arm/microchip_mec/mec1501/Kconfig.soc (hunk 3 above)
Comment (scottwcpg) [reply soc/arm/microchip_mec/mec1501/Kconfig.soc]: Done. Should be original version as I have not changes other than initial mistake.

**Comment by scottwcpg** [comment]
Removed code from soc.c that isn't part this change and causing Kconfig issue.

**Review by dcpleung — APPROVED** [review]

**Comment by fabled** [comment]
Does this driver actually work? To me it looks like in the transaction the TX buffers are handled first, and then RX buffers. Where as to my understanding they are to be handled simultaneously.

**Pull request merged**